### PR TITLE
17947 - Multiple passes of quiz by assignee (number of try and dueTime display)

### PR DESCRIPTION
**Pull request by artemT82**
screenshot from 2016-03-24 00 35 51

### CODE REVIEW

**Comment by krambertech** [comment]
Please, make a screenshot for mobile divice

File: shared/components/pages/ActivationPage.jsx [hunk 1]
```diff
@@ -236,19 +240,45 @@ export default class ActivationPage extends React.Component {
...
                                 </span>
+
+                                {
+                                    dueTime
+                                    ?   (
+                                        <span className='ActivationPage__expires'>
+                                            <span className='ActivationPage__span-divider'>
+                                                •
+                                            </span>
+                                            <span>{l('expires ')}{getTimeFromNow(dueTime)}</span>
```
Comment (krambertech) [inline shared/components/pages/ActivationPage.jsx]: You should use sprints syntax here.

```
sprintf(l('expires %s'), getTimeFromNow(dueTime))
```

File: shared/components/pages/ActivationPage.jsx [hunk 2]
```diff
@@ -183,10 +184,13 @@ export default class ActivationPage extends React.Component {
             numberOfQuestions,
             timeToPass,
             author,
-            isSponsored
+            isSponsored,
+            canAssigneePass,
```
Comment (krambertech) [inline shared/components/pages/ActivationPage.jsx]: Is it working OK id assignee id is not passed?

File: shared/components/pages/ActivationPage.jsx [hunk 3]
```diff
@@ -236,19 +240,45 @@ export default class ActivationPage extends React.Component {
...
+                                                onClick   = {onPass.bind(null, activation)}
+                                                className = 'ActivationPage__btn ActivationPage__pass-btn'
+                                            >
+                                            {l('Pass the test')}
+                                            </Button>
+                                            {
+                                                numberOfTriesLeft
+                                                ?
+                                                    <div className='ActivationPage__number-of-tries'>
+                                                        {l('Number of tries left: ')}{numberOfTriesLeft}
```
Comment (krambertech) [inline shared/components/pages/ActivationPage.jsx]: I suppose, it will be better to say

`You have %d tries left `

... or something like that

**Comment by artemT82** [comment]
Here is how "expires ..." looks on mobile devices
screenshot from 2016-03-24 17 33 06

File: shared/components/pages/ActivationPage.jsx [hunk 4]
```diff
@@ -183,10 +184,15 @@ export default class ActivationPage extends React.Component {
...
+            isSponsored,
+            canAssigneePass,
+            numberOfTriesLeft,
+            dueTime
         } = activation;
 
-        const { l, ngettext, humanizeDuration } = this.context.i18n;
+        const { l, ngettext, humanizeDuration, getTimeFromNow } = this.context.i18n;
+
+        const isPassingBtnAvailable = (typeof canAssigneePass === 'undefined') ? true : canAssigneePass;
```
Comment (krambertech) [inline shared/components/pages/ActivationPage.jsx]: Please, rewrite this check

File: shared/components/pages/ActivationPage.jsx [hunk 5]
```diff
@@ -236,19 +244,51 @@ export default class ActivationPage extends React.Component {
...
-                                        {l('Pass the test')}
-                                        </Button>
+                                        <div className='ActivationPage__pass-btn-wrapper'>
+                                            <Button
+                                                ripple
+                                                raised    = {!isSponsored}
+                                                onClick   = {onPass.bind(null, activation)}
+                                                className = 'ActivationPage__btn ActivationPage__pass-btn'
+                                            >
+                                            {l('Pass the test')}
```
Comment (krambertech) [inline shared/components/pages/ActivationPage.jsx]: Add indent here

File: shared/components/pages/ActivationPage.jsx (hunk 2 above)
Comment (artemT82) [inline shared/components/pages/ActivationPage.jsx]: Totally

**Pull request merged**